lower.ts

```diff
@@ -3,26 +3,26 @@ import * as IR from './ir';
 import { Type, SourceLocation } from './ast';
 import { GlobalEnv } from './compiler';
 
-const nameCounters : Map<string, number> = new Map();
+const nameCounters : Map<string, bigint> = new Map();
```

Comment:
Again, this change suggests that we have programs with more than `2^32` (or even more) variables!

Reply:
Hi Joe,

  Thanks for the questions. I have fixed the above code. Initially I make all numbers javaScript big numbers because I don't want to be confused by when to use numbers and when to use big numbers, but big numbers are indeed overkill in these cases and could cause confusion for others who don't understand our design.

Sincerely,
Zack